src/main/java/com/autotune/analyzer/deployment/KruizeDeployment.java

```diff
@@ -335,15 +337,24 @@ private static KruizeObject getAutotuneObject(String autotuneObjectJsonStr) {
             JSONObject sloJson = null;
             String slo_class = null;
             String direction = null;
-            String objectiveFunction = null;
+            JSONObject objectiveFunctionJson = null;
+            String objFuncType = null;
+            String expression = null;
             String hpoAlgoImpl = null;
             if (specJson != null) {
                 sloJson = specJson.optJSONObject(AnalyzerConstants.AutotuneObjectConstants.SLO);
                 slo_class = sloJson.optString(AnalyzerConstants.AutotuneObjectConstants.SLO_CLASS);
                 direction = sloJson.optString(AnalyzerConstants.AutotuneObjectConstants.DIRECTION);
                 hpoAlgoImpl = sloJson.optString(AnalyzerConstants.AutotuneObjectConstants.HPO_ALGO_IMPL,
                         AnalyzerConstants.AutotuneObjectConstants.DEFAULT_HPO_ALGO_IMPL);
-                objectiveFunction = sloJson.optString(AnalyzerConstants.AutotuneObjectConstants.OBJECTIVE_FUNCTION);
+                objectiveFunctionJson = sloJson.optJSONObject(AnalyzerConstants.AutotuneObjectConstants.OBJECTIVE_FUNCTION);
+                objFuncType = objectiveFunctionJson.optString(AnalyzerConstants.AutotuneObjectConstants.OBJ_FUNCTION_TYPE);
+                if (objFuncType.equals("expression"))
+                    expression = objectiveFunctionJson.optString(AnalyzerConstants.AutotuneObjectConstants.EXPRESSION);
+                else {
+                    if (!objFuncType.equals("source"))
+                        throw new InvalidValueException("Objective function type can only be either 'expression' or 'source' ");
```

Comment:
This should ideally be done in the validate function. Please move it there

Reply:
It can also have other types of checks, such as what happens if the `type` says `expression` and there is no `expression` field. Similarly, `type` says `source`, but we have an `expression` field as well etc

Reply:
Moved